src/librustdoc/passes/mod.rs

```diff
@@ -336,18 +338,22 @@ pub fn look_for_tests<'tcx>(
     find_testable_code(&dox, &mut tests, ErrorCodes::No);
 
     if check_missing_code == true && tests.found_tests == 0 {
+        let mut sp = span_of_attrs(&item.attrs);
```

Comment:
this + the if below can be

```rust
let sp = span_of_attrs(&item.attrs).substitute_dummy(item.source.span());
```

https://doc.rust-lang.org/nightly/nightly-rustc/syntax_pos/struct.Span.html#method.substitute_dummy

Reply:
Oh awesome!